NetTally/ReorderTasksWindow.xaml.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using NetTally.Utility;
+using NetTally.ViewModels;
+
+namespace NetTally
+{
+    /// <summary>
+    /// Interaction logic for ReorderTasks.xaml
+    /// </summary>
+    public partial class ReorderTasksWindow : Window, INotifyPropertyChanged
+    {
+        #region Constructor and variables
+        public ListCollectionView TaskView { get; }
+
+        MainViewModel MainViewModel { get; }
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        public ReorderTasksWindow()
+        {
+            InitializeComponent();
+        }
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mainViewModel">The primary view model of the program</param>
+        public ReorderTasksWindow(MainViewModel mainViewModel)
+        {
+            InitializeComponent();
+
+            MainViewModel = mainViewModel;
+
+            MainViewModel.PropertyChanged += MainViewModel_PropertyChanged;
+
+            // Create filtered, sortable views into the collection for display in the window.
+            if (MainViewModel.TaskList != null)
+            {
+                TaskView = new ListCollectionView(MainViewModel.TaskList);
+            }
+
+            // Initialize starting selected positions
+            TaskView.MoveCurrentToPosition(-1);
+
+            // Set the data context for binding.
+            DataContext = this;
+
+            // Update the displayed order immediately
+            UpdateTaskList();
+        }
+        #endregion
+
+        private void UpdateTaskList()
+        {
+            VoteCounter.Instance.OrderedTaskList = TaskView.SourceCollection as List<string>;
+            OnPropertyChanged("Votes");
```

Comment:
OK, `ReorderTasksWindow` has a `ListView`.  That list view shows what's in a list (`MainViewModel.TaskList`). `MainViewModel.TaskList` should in turn simply redirect to `VoteCounter.OrderedTaskList` because that's the part of the data model that it's exposing.  That means that any changes to `VoteCounter.OrderedTaskList` automatically get reflected in the `ListView`.  That in turn means you shouldn't be constantly replacing the `OrderedTaskList`.  You should simply have a function in the `MainViewModel` that you call when you want to have it modified.  EG:
```
public void IncreaseTaskPosition(int currentPosition) {}
public void DecreaseTaskPosition(int currentPosition) {}
public void ResetTasksOrder(TasksOrdering order) {}
```
And maybe a function that takes the task name instead of the position.  TasksOrdering would be an enum with `Alphabetical` and `AsTallied` (or whatever makes the meaning clear).

From there, the `ReorderTasksWindow` only sends the reordering commands to the `MainViewModel`, which handles modifying the underlying list, which will automatically get propagated back to the `ListView`.

Any events that get raised happen naturally when a data property is changed.  Since the ReorderTasksWindow has no properties, it wouldn't be raising any events on its own.